test/unit/services/url.ts

```diff
@@ -526,6 +526,24 @@ suite('URL Service', ({ expect, spy, stub, itShouldBeCore, itShouldExtendBaseSer
         url
       );
     });
+
+    it('should handle errors thrown by `replaceState`', () => {
+      const err = new Error('Whoops, something went wrong!');
+      const replaceState = stub().throws(err);
+      const store = { getState: spy() };
+      const warn = spy();
+      win.document = { title: 'Foo' };
+      win.history = { replaceState };
+      stub(service, 'filterState');
+      service['app'] = <any>{
+        log: { warn },
+        flux: { store },
+      };
+
+      service.replaceHistory('/foo');
+
+      expect(warn).to.have.been.calledWith('unable to replace browser history', err);
```

Comment:
W generally only need to expect that the function was called in the case of a logger because the message is incidental. If you feel like you should still test that it was called with certain parameters, you can include `sinon` where you include `expect` and `spy` and the others on line 11 and replace this string with [`sinon.match.string`](https://sinonjs.org/releases/v6.1.5/matchers/#sinonmatchstring-1).